CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+[tidyteam code review prinicples](https://code-review.tidyverse.org/issues/).
+
+Once the foreseen edits are documented in an issue, create a pull request
+(to see how simply follow
+[Github's instructions on creating a Pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request))
+and add your changes to the related branch.
+
+## Development
+
+We use [renv](https://rstudio.github.io/renv/index.html) for our development
+environment and would encourage you to do the same.
+Therefore, abn directly provides a development environment and we suggest the
+following workflow for development:
+
+*   Fork the package and clone it onto your computer.
+    
+    _Note:_ You might do this with `usethis::create_from_github("furrer-lab/abn", fork = TRUE)`.
+
+*   Install `renv` with `install.packages("renv")`
+*   Restore the development environment with `renv::restore()`
+*   Create a new branch for your pull request.
+    
+    _Note:_ You might use `usethis::pr_init("explicative-branch-name")`.
+    
+    _See [Testing](#testing) for flags you can set in the branch name_
+
+*   Edit the code and commit your changes.
+  
+    _Note:_ With `usethis` you would run: `usethis::pr_push()` and the follow 
+    the prompts.
+    The title of your pull request should briefly describe the change.
+    The body of your pull request should contain `Closes #issue-number`, where
+    `#issue-number` is the identifier of the related issue.
+
+## Testing
+
+The abn package uses several testing pipelines.
+When developing (i.e. not on the `main` branch) we recommend running the
+[quick-testthat.yml](https://github.com/furrer-lab/abn/blob/24-documentation-of-the-testing-procedure-noT/.github/workflows/quick-testthat.yml) action.
+This pipeline runs be default on push event on all branches other than `main`
+and executes all the test defined withing the abn package by calling
+[`devtools::test`](https://devtools.r-lib.org/reference/test.html).
+It does so in a docker container from the
+[r-containers](https://github.com/furrer-lab/r-containers) package which builds
+container images that are setup with all the required dependencies.
+
+By default, a container based on `Debian` using `clang` as compiler front-end
+and the current R development version version (`debian/clang/devel`) is used.
+However, [other configurations are available](https://github.com/orgs/furrer-lab/packages?repo_name=r-containers).
+
+If you want to change the container image to use for testing your branch against,
+you can append the desired configuration to the name of your branch with the
+prefix `'__'`.
+Hence, a branch named `my-awesome-branch__debian/gcc/release` would use the
+container image based on Debian, using `gcc` and the current release version of
+R to run the tests.
+
+You can also completely opt-out of testing by adding the flag `noT` to your
+branch name.
+
```

Comment:
Maybe add an example here, too. I am regularly unsure if `noT` can be anywhere in the branch name or if it must be at the end separated by double underscores, like for the container specification. The same is true for the commit message.

Reply:
Good point. Currently it can be anything which actually bares the risk of falsely detecting the flag in CamelCase names like "monoTestBranch".

Reply:
https://github.com/furrer-lab/abn/pull/82/commits/420ae5f828d1fbd8ac49e272953d85cd7b14f390